src/prefect/client/orchestration.py

```diff
@@ -2526,6 +2531,48 @@ async def delete_resource_owned_automations(self, resource_id: str):
 
         await self._client.delete(f"/automations/owned-by/{resource_id}")
 
+    async def acquire_concurrency_slots(
+        self, names: List[str], slots: int
+    ) -> List[MinimalConcurrencyLimitResponse]:
+        try_count = 0
+        while True:
+            retry_seconds = None
+            try_count += 1
+            try:
+                response = await self._client.post(
+                    "/v2/concurrency_limits/increment",
+                    json={"names": names, "slots": slots},
+                )
+            except Exception as exc:
+                if exc.response.status_code == 423:
```

Comment:
my hunch is that this sort of retry logic would be better defined within the context manager so that the client remains "just" a wrapper to the API - wdyt?

Reply:
Maybe we want two separate functions here, like a `try-and-fail-quickly` version and a `wait-until-i-can-proceed` version.  The `wait-until-i-can-proceed` version would call the `try-and-fail-quickly` in the retry/backoff loop, and the context manager would use the `wait-until-i-can-proceed` version.  That way most users can use the context manager for clarity, but advanced users can use either the `wait-until-i-can-proceed` version or the `try-and-fail-quickly` version to build their own synchronization primitives.

Reply:
That's a good point, and I like making it a couple of utility functions that users can build upon. I'll make it so.